todo/src/db/connect.rs

```diff
@@ -0,0 +1,20 @@
+use std::sync::Arc;
+
+use http_problem::prelude::*;
+use tokio_postgres::Client;
+
+pub async fn connect() -> Result<Arc<Client>> {
+    let (client, connection) = tokio_postgres::connect(
+        "host=localhost user=postgres password=postgres dbname=onboarding",
+        tokio_postgres::NoTls,
+    )
+    .await?;
+
+    tokio::spawn(async move {
+        if let Err(err) = connection.await {
+            println!("connection error: {err}")
+        }
+    });
+    println!("Successfully connected to database");
+    Ok(Arc::new(client))
```

Comment:
Aqui você spawna uma task pra validar se a conexão tá OK, mas diz fora da task que a conexão foi criada com sucesso. Mesmo com `connection.await` resultando em erro, `connect()` vai retornar um `Ok` porque o teste foi feito numa outra task. Não seria melhor fazer sem spawnar uma task?

```suggestion
    connection.await?; // retornar o erro caso `connection.await` falhe
    println!("Successfully connected to database");
    Ok(Arc::new(client))
```

Reply:
Tem que ser em outra task porque a Future só resolve quando a conexão fecha. https://docs.rs/tokio-postgres/latest/tokio_postgres/struct.Connection.html

Reply:
Sobre o `println!("Successfully connected to database");` eu concordo.

Reply:
Qual é o objetivo dessa task então?

Reply:
Eh, desse jeito que vc sugeriu fica hanging forever @saiintbrisson 
> Qual é o objetivo dessa task então?
como assim?

Reply:
Que que vocês querem atingir fazendo await na conexão? Só pegar caso de erro fatal?

Reply:
Concordo que não é o mais ideal, mas aqui a ideia da task é só dar o feedback se a conexão tiver fechado por algum problema.

Como não usamos o tokio-postgres diretamente assim nos serviços, acho que não precisamos complicar muito aqui.